templates/todo/api/java/src/main/resources/application.properties

```diff
@@ -3,11 +3,13 @@ server.port=3100
 spring.jackson.date-format=com.microsoft.azure.simpletodo.configuration.RFC3339DateFormat
 spring.jackson.serialization.WRITE_DATES_AS_TIMESTAMPS=false
 
+# Temporally disabling auto-configuration for key vault
```

Comment:
What's the plan for re-enabling this?  Might be nice to have a bug in our repository and link here to that tracking issue.

My guess here is there's no similar trick we can play like we did with `MongoClientSettingsBuilderCustomizer` to customize the KeyClient that would be used by some spring component to inject this stuff, so we are just turning it all off.

But I'd like to better understand what it would mean for us to try to support this in the future?

Reply:
For re-enabling, we would just remove line `10`, which is setting the key vault url and the name of the secret that holds the mongo db connection string, and uncomment the lines that this PR is commenting.
That's why I didn't want to remove the lines, and just disable them. 

As soon as AzdCredential is shipped in the AzureSDK identity library and it is part of the AzureDefaultCredential, we would just need to revert all the changes from the PR:
- Remove the AzdCredential
- Uncomment the application.properties to let spring to pull the secrets from key vault automatically and let mongo use the property for connection string
- Remove the bean to manually set the connection string for the mongoDb

Reply:
Got it.  Sort of a bummer that this stuff is all tied to DefaultAzureCredential instead of allowing folks to provide any instance of TokenCredential (if we had that ability, we could just put our chained token credential) in there - but I guess that's the lay of the land in Spring world.

Reply:
I started a conversation with the Spring-team, @stliu is working on implementing something which will allow us to add our credential for the auto-configuration. Then we wouldn't have to disable it.